lib/Sema/CodeSynthesis.cpp

```diff
@@ -1008,11 +1008,18 @@ InheritsSuperclassInitializersRequest::evaluate(Evaluator &eval,
   assert(superclass);
 
   // If the superclass has known-missing designated initializers, inheriting
-  // is unsafe.
+  // is unsafe if there is any additional storage defined in the client.
   auto *superclassDecl = superclass->getClassOrBoundGenericClass();
   if (superclassDecl->getModuleContext() != decl->getParentModule() &&
-      superclassDecl->hasMissingDesignatedInitializers())
-    return false;
+      superclassDecl->hasMissingDesignatedInitializers()) {
+    return llvm::none_of(decl->getMembers(), [](Decl *member) {
```

Comment:
Don't we still want to perform the rest of the checks below in the case that this returns `true` (i.e the subclass doesn't introduce any storage)? For example, I think we still want to make sure that a subclass that defines its own designed init but doesn't introduce storage doesn't inherit convenience inits.

Reply:
Yeah, this solution is broken.  We need to sit down and rethink these rules before rushing out a fix like this. Closing this.

Reply:
But, for context, the overriding half of the rules currently don’t fire if you have missing designated inits, so this is a strictly additive change with respect to the current semantics. Even though those semantics are bad.